repos/spack_repo/builtin/packages/py_scifem/package.py

```diff
@@ -39,11 +40,15 @@ class PyScifem(PythonPackage):
     depends_on("py-scikit-build-core+pyproject", type="build")
     depends_on("py-setuptools@42:", type="build")
     depends_on("cmake@3.21:", type="build")
+    depends_on("fenics-dolfinx@0.9:", when="@0.4:", type=("build", "link"))
+    depends_on("fenics-dolfinx@main", when="@main", type=("build", "link"))
+    depends_on("fenics-basix@0.9:", when="@0.4:", type=("build", "link"))
+    depends_on("fenics-basix@main", when="@main", type=("build", "link"))
```

Comment:
```suggestion
    depends_on("fenics-dolfinx@main", when="@main", type=("build", "link"))
    depends_on("fenics-dolfinx@0.9:", when="@0.4:", type=("build", "link"))
    depends_on("fenics-basix@main", when="@main", type=("build", "link"))
    depends_on("fenics-basix@0.9:", when="@0.4:", type=("build", "link"))
```

Let's put `main` above v0.9 since it is classified as an infinite version.